### DISCUSSION TITLE
nginx setup on Docker Compose Windows to serve Immich

### DISCUSSION THREAD

**Discussion by herculespt** [NginxProxyManager/nginx-proxy-manager#3777] [General]
Hi,

Can anyone help me setting up a reverse proxy so I can achieve my goal?

I have setup Immich and running. I have setup Nginx and running. It's been almost 6 months I search for a solution to use nginx as a reverse proxy so I don't need to change the url address in Immich when I am at my  Home Immich network.

I have a DDNS setup, with an URL assigned, I have my router with the corresponding ports exposed.

The problem is:

When I am in my home network, I have to use the localhost IP and Port  to access Immich instead of the URL I use while I'm on an external network. I want to use that URL when I am at home network, instead of the localhost IP.

Any help would be appreciated.

**Comment by Glowazowski** [comment]
This sounds like a problem related to DNS rebind protection of your router.

I have setup my own DNS (AdGuard Home - Pi-hole works, too) and defined a DNS-rewrite: immich.<my-domain> now points directly to the IP of my Immich instance. For me, this solved the problem you described.
